lib/hmr/JavascriptHotModuleReplacement.runtime.js

```diff
@@ -107,9 +107,9 @@ module.exports = function () {
 		var outdatedModules = [];
 		var appliedUpdate = {};
 
-		var warnUnexpectedRequire = function warnUnexpectedRequire() {
+		var warnUnexpectedRequire = function warnUnexpectedRequire(moduleId) {
 			console.warn(
-				"[HMR] unexpected require(" + result.moduleId + ") to disposed module"
+				"[HMR] unexpected require(" + moduleId + ") to disposed module"
 			);
 		};
```

Comment:
As `warnUnexpectedRequire` is a module factory it's parsed a `module` argument, which has an `id` property.

``` js
		var warnUnexpectedRequire = function warnUnexpectedRequire(module) {
			console.warn(
				"[HMR] unexpected require(" + module.id + ") to disposed module"
			);
		};
```